src/net_processing.cpp

```diff
@@ -673,46 +698,9 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
 
         return true; // return true so we don't get any process message failures in the log.
     }
-    // BUVERSION is used to pass BU specific version information similar to NetMsgType::VERSION
-    // and is exchanged after the VERSION and VERACK are both sent and received.
-    else if (strCommand == NetMsgType::BUVERSION)
-    {
```

Comment:
Removing the BU messages, before all peers have upgraded, prevents non-upgraded nodes from using expedited forwarding and prevents us from forwarding to those peers as well. Usually we would remove old version code after we release and have passed the next hard fork date to make sure everyone has upgraded.  But, it would be a pain though to put all this code back in for a feature, which to my knowledge, isn't really used much.

Reply:
buversion hasnt been used by bu nodes since xversion was implemented. any node still using buversion message over xversion is not on the current consensus rules anyway.

Reply:
oh good